intranet/apps/eighth/models.py

```diff
@@ -824,28 +826,30 @@ def get_viewable_members_serializer(self, request):
         Returns: QuerySet
 
         """
-        ids = []
         user = request.user
-        for member in self.members.all():
-            show = member.can_view_eighth or (user and (user.is_eighth_admin or user.is_teacher or member == user))
-
-            if show:
-                ids.append(member.id)
+        if user and (user.is_eighth_admin or user.is_teacher):
+            return self.members.all()
+        else:
+            ids = []
+            for member in self.members.all():
```

Comment:
You could try to offload a bit of work to the DB:
```python
for member in self.members.exclude(id=user.id).all():
    if member.can_view_eighth:
        ids.append(member.id)
```

Reply:
That wouldn't give the required behavior -- if the user is signed up, they should be included too. I'd have to make another query to see if they're signed up and add them separately, which kind of defeats the point.